src/emulator/kernel/include/kernel/thread/thread_state.h

```diff
@@ -44,5 +46,5 @@ struct ThreadState {
     std::mutex mutex;
     std::condition_variable something_to_do;
     std::vector<std::shared_ptr<ThreadState>> waiting_threads;
-    char name[32];
+    std::string name;
```

Comment:
There was some memory corruption bug related to the name. You're probably right that it would be a simple fix without going for `std::string` but it was crashing most runs for me and I needed to work around it.